CORE-5921 Align signature file naming behavior in cli package plugin with `jarsigner`

kyriathar opened the pull request:
Aligns signature files (.SF, .RSA) naming with `jarsigner`  across all package module any time we need to sign a Cpx.

- adds `--sig-file` option to specify signature files base name for all commands that do signing.
- if `--sig-file` is missing then key alias is used, modified if needed as per [jarsigner docs](https://docs.oracle.com/javase/7/docs/technotes/tools/windows/jarsigner.html#:~:text=a%20secure%20system.-,%2Dsigfile%20file,-Specifies%20the%20base).

josephzunigadaly commented on tools/plugins/package/src/test/kotlin/net/corda/cli/plugins/packaging/TestUtils.kt:
```diff
@@ -137,4 +137,20 @@ internal object TestUtils {
+
+    fun jarEntriesExistInCpx(cpxPath: Path, expectedEntries: List<String>): Boolean {
+        JarInputStream(Files.newInputStream(cpxPath)).use {
+            val jarEntries = mutableListOf<JarEntry>()
+            var jarEntry: JarEntry? = it.nextJarEntry
+            while (jarEntry != null) {
+                jarEntries.add(jarEntry)
+                jarEntry = it.nextJarEntry
+            }
+
+            val actualEntries = jarEntries.map { it.name }
```
```suggestion
            val actualEntries = generateSequence { it.nextJarEntry?.name }.toList()
```

kyriathar-r3 committed:
CORE-5921 Align signature file naming behavior in cli package plugin with jarsigner (#1759)

Aligns signature files (.SF, .RSA) naming with jarsigner across all package module any time we need to sign a Cpx.

- adds `--sig-file` option to specify signature files base name for all commands that do signing.
- if `--sig-file` is missing then it uses key alias as signature files base name, modified if needed as per jarsigner docs.
